### PR TITLE
login & logout with token

### PR DESCRIPTION
**Pull request by AlexMegas**
Did refactoring for "signup", "login", "logout".
"login", "logout" - works with token.

### CODE REVIEW

File: controllers/user.js [hunk 1]
```diff
@@ -4,50 +4,98 @@ const randomToken = require('random-token')
...
+            message: 'User logged in',
+            token: user.token
+        }))
+        .catch((err) => console.error(err))
 };
 
 
 exports.logout = (req, res) => {
-    return res.status(200).send('User logged out :)');
+    const authHeader = req.get('Authorization');
```
Comment (IBurychka) [inline controllers/user.js:79]: Now we send token in token field in the header. Change req.get('Authorization') to req.headers.token

File: controllers/user.js [hunk 2]
```diff
@@ -4,50 +4,98 @@ const randomToken = require('random-token')
...
+    if (!authHeader) {
```
Comment (IBurychka) [inline controllers/user.js:80]: change authHeader to token

File: controllers/book.js [hunk 3]
```diff
@@ -76,8 +90,10 @@ exports.updateBookById = (req, res) => {
 
 
 exports.deleteBookById = (req, res) => {
-    const bookId = req.params.id;// find book by id and delete
-    Book.findOneAndDelete({ _id: bookId })
+    const bookId = req.params.id; // find book by id and delete
+    Book.findOneAndDelete({
+            _id: bookId
+        })
         .then(res.send(`Record with id:${bookId} --> deleted from DB!`))
```
Comment (IBurychka) [inline controllers/book.js]: better to send book title instead of book id

File: routes/book.js [hunk 4]
```diff
@@ -7,9 +7,13 @@ const {
...
+    checkToken
+} = require('../middlewares/auth.js');
+
 const book = express.Router();
 
-book.post('/', addBook); // record new book
+book.post('/', checkToken, addBook); // record new book
 book.get('/:id', getBookById); // select one book by its id
 book.get('/', getBookByQuery); // select book by title or all
 book.patch('/:id', updateBookById); // update book parametres by id
```
Comment (IBurychka) [inline routes/book.js]: also you need to add checkToken to delete book

File: controllers/user.js (hunk 1 above)
Comment (AlexMegas) [reply controllers/user.js:79]: changed.

File: controllers/user.js (hunk 2 above)
Comment (AlexMegas) [reply controllers/user.js:80]: changed.

File: controllers/book.js (hunk 3 above)
Comment (AlexMegas) [reply controllers/book.js]: changes applied.

File: routes/book.js (hunk 4 above)
Comment (AlexMegas) [reply routes/book.js]: In branch "token" token checks are applied.

**Pull request closed without merging**